app/src/main/java/com/example/mytestproject/ui/weatherData/dailyWeather/ThreeDaysWeatherFragment.kt

```diff
@@ -0,0 +1,73 @@
+package com.example.mytestproject.ui.weatherData.dailyWeather
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.mytestproject.App
+import com.example.mytestproject.R
+import com.example.mytestproject.util.showDailyWeatherRequestResult
+import kotlinx.android.synthetic.main.fragment_three_days_weather.*
+import javax.inject.Inject
+
+
+class ThreeDaysWeatherFragment : Fragment() {
+
+    @Inject
+    lateinit var dailyWeatherFactory: DailyWeatherFactory
+    private lateinit var dailyWeatherViewModel: DailyWeatherViewModel
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_three_days_weather, container, false)
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        (activity?.applicationContext as App).weatherDataComponent.inject(this)
+
+        setTitle("Moscow")
+
+        dailyWeatherViewModel =
+            ViewModelProvider(this, dailyWeatherFactory).get(DailyWeatherViewModel::class.java)
+
+        dailyWeatherViewModel.getWeatherData(4) // one more day because the first current day is not included in the list
+
+        context.let {
```

Comment:
Why with let function?

Reply:
I saw it somewhere,  because the context can be null

Reply:
On this method context shouldn't be null, because activity was created. Fragment is attached to activity already. 
Instead of `context` we can use `requireContext()`.

Reply:
Fixed